Patch from upstream:
http://strace.cvs.sourceforge.net/viewvc/strace/strace/linux/arm/syscallent.h?r1=1.17&r2=1.18&sortby=file

2008-11-13  Kirill A. Shutemov  <kirill@shutemov.name>

    * linux/arm/syscallent.h: Fix build on ARM EABI which does not
    provide syscalls socketcall and ipc.

--- strace-4.5.18.orig/linux/arm/syscallent.h	2008/12/29 20:04:15	1.17
+++ strace-4.5.18/linux/arm/syscallent.h	2009/01/01 23:20:38	1.18
@@ -431,6 +431,7 @@
 	{ 5,	0,	printargs,		"SYS_398"	}, /* 398 */
 	{ 5,	0,	printargs,		"SYS_399"	}, /* 399 */
 
+#ifndef __ARM_EABI__
 #if SYS_socket_subcall != 400
  #error fix me
 #endif
@@ -481,3 +482,4 @@
 	{ 4,	TI,	sys_shmdt,		"shmdt"		}, /* 440 */
 	{ 4,	TI,	sys_shmget,		"shmget"	}, /* 441 */
 	{ 4,	TI,	sys_shmctl,		"shmctl"	}, /* 442 */
+#endif
